civis/tests/test_io.py

```diff
@@ -536,102 +536,120 @@ def test_civis_file_to_table_table_does_not_exist_some_sql_types_missing(
     @mock.patch('civis.io._tables._process_cleaning_results')
     @mock.patch('civis.io._tables._run_cleaning')
     def test_civis_file_to_table_table_doesnt_exist_provide_table_columns(
-            self,
-            m_run_cleaning,
-            m_process_cleaning_results,
-            _m_get_api_spec
+        self,
+        m_run_cleaning,
+        m_process_cleaning_results,
+        _m_get_api_spec
     ):
-        table = "scratch.api_client_test_fixture"
-        database = 'redshift-general'
-        mock_file_id = 1234
-        mock_cleaned_file_id = 1235
-        mock_import_id = 8675309
 
-        self.mock_client.imports.post_files_csv.return_value\
-            .id = mock_import_id
-        self.mock_client.get_database_id.return_value = 42
-        self.mock_client.default_credential = 713
-
-        self.mock_client.get_table_id.side_effect = ValueError('no table')
-        table_columns = [{'name': 'foo', 'sql_type': 'INTEGER'},
-                         {'name': 'bar', 'sql_type': 'VARCHAR(42)'}]
-        m_process_cleaning_results.return_value = (
-            [mock_cleaned_file_id],
-            True,  # headers
-            'gzip',  # compression
-            'comma',  # delimiter
-            None  # table_columns
-        )
-        m_run_cleaning.return_value = [mock.sentinel.cleaning_future]
-
-        with mock.patch.object(
-                civis.io._tables, 'run_job',
-                spec_set=True) as m_run_job:
-
-            run_job_future = mock.MagicMock(
-                spec=civis.futures.CivisFuture,
-                job_id=123,
-                run_id=234
+        # pytest.parametrize apparently doesn't work with unittest.TestCase
+        # objects, which this ultimately inherits from, so we'll define a
+        # function here and run subtests below.
+        # See https://docs.pytest.org/en/stable/unittest.html.
+        def run_subtest(mock_file_ids):
+            table = "scratch.api_client_test_fixture"
+            database = 'redshift-general'
+            mock_cleaned_file_ids = mock.Mock()
```

Comment:
I changed this to just be a mock rather than an int, to avoid having to add logic about how long the list was, etc.